src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -338,6 +351,11 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 	}
 
 	private ExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+		ITypeSymbol? returnTypeInfo = Model.GetTypeInfo( invocationExpr ).Type;
+		if( returnTypeInfo?.MetadataName == "IAsyncEnumerable`1" && returnTypeInfo.ContainingNamespace.ToString() == "System.Collections.Generic" ) {
+			m_removeAsyncAnywhere = true;

```

Comment:
I think we probably want to store the previous value of `m_removeAsyncAnywhere` (it could be true or false here) and restore it on the way back out of the function?